core/src/test/java/org/apache/iceberg/variants/TestShreddedArray.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.variants;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.List;
+import java.util.Random;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.RandomUtil;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public class TestShreddedArray {
+  private static final VariantMetadata EMPTY_METADATA =
+      Variants.metadata(VariantTestUtil.emptyMetadata());
+
+  private static final List<VariantValue> ELEMENTS =
+      ImmutableList.of(
+          Variants.of(34), Variants.of("iceberg"), Variants.of(new BigDecimal("12.21")));
+
+  private final Random random = new Random(871925);
+
+  @Test
+  public void testShreddedFields() {
```

Comment:
I think this test name is based on the `ShreddedObject` version, but doesn't really fit here. That test suite distinguishes between fields that are _unshredded_ (stored in the encoded/serialized object) and _shredded_ (stored in an in-memory map, as with reconstructed shredded fields). But arrays don't allow partial shredding so there are no unshredded elements. Also, this is testing the basic functionality of `get` with elements and not fields.

For this, I'd just rename the test case to `testElementAccess()`.